### PR TITLE
Added monodevelop like add file dialog to Gtk Pipeline

### PR DESCRIPTION
**Pull request by cra0zy**
addfiledialog

Notes:
<ul>
<li>Fixed undoing "add file" action</li>
<li>When adding a file this dialog will show up</li>
<li>Windows Pipeline will act like usual</li>
<li>Undo/Redo can undo adding a file, but will not delete copied files, I leave that to the user.</li>
</ul>

### CODE REVIEW

**Comment by tomspilman** [comment]
@cra0zy - I fixed your image link.

**Comment by tomspilman** [comment]
> Undo/Redo can undo adding a file, but will not delete copied files, I leave that to the user.

That seems reasonable.

File: Tools/Pipeline/Windows/MainView.cs [hunk 1]
```diff
@@ -901,5 +901,12 @@ public void ItemExistanceChanged(IProjectItem item)
                 }
             }
         }
+
+        public bool CopyOrLink(string file, bool exists, out int def, out bool applyforall)
+        {
+            def = 2;
+            applyforall = true;
+            return true;
```
Comment (tomspilman) [inline Tools/Pipeline/Windows/MainView.cs]: @cra0zy - So this is where someone would hook in the dialog for Windows?

What is `def`?  Can you rename that to something more obvious?

**Comment by tomspilman** [comment]
I think we had an issue for this somewhere... we should find and link it.

File: Tools/Pipeline/Windows/MainView.cs (hunk 1 above)
Comment (cra0zy) [inline Tools/Pipeline/Windows/MainView.cs]: I'll rename it, def is default

**Comment by cra0zy** [comment]
Thanks for fixing the image link!!!

File: Tools/Pipeline/Windows/MainView.cs (hunk 1 above)
Comment (tomspilman) [inline Tools/Pipeline/Windows/MainView.cs]: Default what?  Default result?  Default button?

**Comment by cra0zy** [comment]
@tomspilman I think now it's much better

**Comment by tomspilman** [comment]
Yep... much clearer.

**Comment by tomspilman** [comment]
Looks good... going to merge this.  Just need someone to follow up with the implementation for WinForms.

**Pull request merged**